### DISCUSSION TITLE
Remove separators from ItemList controls in the editor for visual consistency with Trees

### DISCUSSION THREAD

**Discussion by davthedev** [godotengine/godot-proposals#9136] [Editor]
There are two controls used to show lists of items: Tree and ItemList.

The most widely used one is Tree. But ItemList is still there and I propose to update its appearance to match Tree controls that have flat lists.

ItemList is used in the Files dock (second half, can switch to icons grid mode) and the code editor. In the current editor theme, there are separators ("guides"?) drawn between items, although there are none between Tree items. I propose to hide those separators in ItemLists to make a more consistent appearance.

Before:

![Capture d’écran du 2024-02-20 01-37-22](https://github.com/godotengine/godot-proposals/assets/26961646/4b3a90c3-6c75-4d4b-8c98-f7ace8948716)

After:

![Capture d’écran du 2024-02-20 01-42-15](https://github.com/godotengine/godot-proposals/assets/26961646/5e631e24-af03-4f19-ad98-8defc96247f9)

![image](https://github.com/godotengine/godot-proposals/assets/26961646/3c12e6d8-7526-4788-8318-286a4af78555)

Pull request: https://github.com/godotengine/godot/pull/88575

**Comment by MewPurPur** [comment]
Agreed, I never liked these separators, especially how they overlap with the selection stylebox.

**Reply by davthedev** [reply]
The glitchy overlap is actually a bug. Submitted a PR to fix  it: https://github.com/godotengine/godot/pull/88577

**Comment by akien-mga** [comment]
Implemented by https://github.com/godotengine/godot/pull/88575.
